internal/cmd/devices/list.go

```diff
@@ -0,0 +1,272 @@
+package devices
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/jedib0t/go-pretty/v6/table"
+	cmds "github.com/saucelabs/saucectl/internal/cmd"
+	"github.com/saucelabs/saucectl/internal/devices"
+	"github.com/saucelabs/saucectl/internal/devices/devicestatus"
+	"github.com/saucelabs/saucectl/internal/tables"
+	"github.com/saucelabs/saucectl/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+const (
+	JSONOutput = "json"
+	TextOutput = "text"
+)
+
+type DeviceWithStatus struct {
```

Comment:
These are fairly generic, might these be better included in `/internal/devices` as a single source for all device related structs, etc?

Reply:
I wouldn't do that, right now this is only specific to this listing. Devices returned from RDC do not normally have status in them, you have to pull that separately.

Reply:
Wouldn't it make sense to pull the function that queries / merges device status out of this then into the devices service then make this generic to share? From what I could tell that seemed to be a common pattern in saucectl.